assignment/src/models/appointment.ts

```diff
@@ -0,0 +1,59 @@
+import Joi from 'joi';
+import mongoose, { Document, Schema } from 'mongoose';
+
+interface IPatient {
+    _id: mongoose.Types.ObjectId;
+    petName: string;
+    ownerPhoneNumber: string;
+}
+
+interface IAppointment extends Document {
+    patient: IPatient;
+    appointmentStartTime: Date;
+    appointmentEndTime: Date;
+    description: string;
+    feeAmount: number;
+    feeStatus: 'USD' | 'EUR' | 'Bitcoin' | 'Unpaid';
+}
+
+const appointmentSchema = new Schema<IAppointment>({
+    patient: { 
+        type: new Schema({
```

Comment:
I think we should use the Patient Id here. ( We can follow the concept of SQL databases )

Reply:
Actually the reason behind using embedded document was to limit the an extra query. Because when fetching appointments we do need patient details (petName, ownerPhoneNumber, ownerAddress) so for them we have to make a query to populate our appointment object. Also as only three properties are embedded in appointment object so i beleive it can be a better approach for efficiency.

Kindly let me know if I'm to make these changes or any other issue that I'm missing out in this.